designs/switch-prometheus-to-opentelemetry.md

```diff
@@ -0,0 +1,172 @@
+Switch Prometheus to OpenTelemetry
+===================
+
+## Motivation
+
+Today, controller-runtime already provides metrics accessible through the
+Prometheus client, and exposed on the `/metrics` endpoint. However, this forces
+components to use Prometheus, when they may want to monitor their app another
+way, not necessarily with a pull system.
+
+## Goals
+
+* Provide a generic way to send metrics, so they can be exported virtually anywwhere.
+* Do not introduce existing metrics changes for users
+
+## Proposal
+
+We will entirely remove the current metrics server. If users need to run
```

Comment:
Huge -1 from me. Having a metric server and not using it doesn't really have much of a drawback. Requiring ppl to write code in order to get any metrics has the huge drawback that they won't be available by default.